app/src/main/java/com/prayertime/utils/LocationUtils.kt

```diff
@@ -0,0 +1,32 @@
+package com.prayertime.utils
+
+import android.content.Context
+import android.location.Location
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.MutableLiveData
+import com.google.android.gms.location.FusedLocationProviderClient
+import com.google.android.gms.location.LocationServices
+
+class LocationUtils{
+
+    private var fusedLocationProviderClient: FusedLocationProviderClient ?= null
+    private var location : MutableLiveData<Location> = MutableLiveData()
+
+    // using singleton pattern to get the locationProviderClient
+    fun getInstance(appContext: Context): FusedLocationProviderClient{
+        if(fusedLocationProviderClient == null)
+            fusedLocationProviderClient = LocationServices.getFusedLocationProviderClient(appContext)
+        return fusedLocationProviderClient!!
+    }
+
+    fun getLocation() : LiveData<Location> {
+        fusedLocationProviderClient!!.lastLocation
+            .addOnSuccessListener {loc: Location? ->
+                location.value = loc
+
+            }
+
+        return location
+    }
+
+}
```

Comment:
@janeiromeosa does this class is used anywhere?

Reply:
Nope. I was just testing out something. I will take it out.